session4/objets/main.js

```diff
@@ -0,0 +1,99 @@
+/* LES OBJETS */
+console.log('Bienvenue dans la console javascript');
+console.log('Vous êtes dans les exercices sur les objets (objects)');
+
+// ------ TODO 1 ------
+
+// 1. Déclarer une variable student qui contient un objet avec les propriétés et valeurs suivantes :
+// - name : Votre nom
+// - age : Votre age
+// - city: Votre ville
+
+
+
+// 4. Afficher dans la console l'objet student.
+
+
+
+
+// ------ TODO 2 ------
+var user = {
+  name: 'John',
+  age: 30
+};
+
+// 1. Ajouter une propriété (clé) "city" de valeur "Paris" à l'objet user.
+
+
+
+// 2. Afficher dans la console l'âge de John.
+
+
+
+// 3. Remplacer la valeur de "city" avec "Nice".
+
+
+
+// 4. Afficher dans la console un tableau des propriétés (clés) de l'objet user grâce à la méthode keys.
+
+
+
+
+// ------ TODO 3 ------
+var users = [user, { name: 'Emma', age: 32, city: 'London' }];
+
+// 1. Afficher dans la console l'objet dont le name est "Emma".
+
+
+
+// 2. Afficher dans la console "John", en accédant aux élements du tableau
+// puis à la propriété de l'objet.
+
+
+
+
+// ------ TODO 4 ------
+var car = {
+  brand: 'Ford',
+  speed: 100
+};
+
+// 1. Ajouter une fonction "drive" à l'objet car qui affiche dans la console la vitesse de l'objet.
```

Comment:
Quand tu dis une fonction, drive à l'objet tu veux dire une méthode ? Si oui, 'this' n'est pas au programme pour le moment. Du coup si tu pensais à faire :

```js
car.drive = function(){ 
  console.log(this.speed);
}
```
Il faut soit rajouter du cours ou remplacer la méthode par un truc plus bateau console.log("Vroom").

Reply:
Oui c'est un bon point ! Je pars sur le Vroom.